core/src/main/kotlin/no/strooped/view/screen/TaskScreen.kt

```diff
@@ -1,3 +1,102 @@
+import com.badlogic.gdx.scenes.scene2d.Stage
+import com.badlogic.gdx.scenes.scene2d.ui.Image
+import com.badlogic.gdx.utils.viewport.ScreenViewport
+import no.strooped.GameSingleton
+import no.strooped.StroopedController
+import no.strooped.TextureSizes
+import no.strooped.model.GameRoom
+import no.strooped.view.screen.components.ColorButton
+import no.strooped.view.screen.components.Label
+
+/**
+ * Uses https://otter.tech/an-mvc-guide-for-libgdx/ as inspiration
+ * */
+private val FONT_SIZE_LABEL_TEXT = TextureSizes.getScaledFontSize(3.0f)
+class TaskScreen(
+    private val controller: StroopedController,
+    private val gameRoom: GameRoom
+) : Screen {
+    private var ui: Stage = Stage(ScreenViewport())
+    private val backgroundPosition = Vector2(0f, 0f)
+    private val background: Image = Image(Texture("white.jpg"))
+    private var colorOptions: Array<ColorButton?> = arrayOfNulls(gameRoom.currentTask!!.possibleAnswers.size)
+    init {
+        print("Stuff")
+        // this init will disappear in future versions
+        /*Label label = new Label(labelText, skin);
+Pixmap labelColor = new Pixmap(labelWidth, labelHeight, Pixmap.Format.RGB888);
+labelColor.setColor(<your-color-goes-here>);
+labelColor.fill();
+label.getStyle().background = new Image(new Texture(labelColor)).getDrawable();*/
+        // TO DO for later
+        /*val toastFactory: Toast.ToastFactory = Builder()
+            .font(attr.font)
+            .build()*/
+        // https://github.com/wentsa/Toast-LibGDX - for toasts
+    }
+
+    override fun hide() {}
+
+    override fun show() {
+        background.setPosition(backgroundPosition.x, backgroundPosition.y)
+        background.setSize(Gdx.graphics.width.toFloat(), Gdx.graphics.height.toFloat())
+        ui.addActor(background)
+        val message = "Task ${gameRoom.currentTask!!.id}"
+        val labelWidth = Gdx.graphics.width * 0.8f
+        val labelPosition = Vector2(
+            (Gdx.graphics.width - labelWidth) * 0.5f,
+            Gdx.graphics.height * 0.85f
+        )
+        val label = Label(message, labelPosition, labelWidth, FONT_SIZE_LABEL_TEXT, Color.BLACK)
+
+        ui.addActor(label)
+        val stringOfColors = GameSingleton.room!!.currentTask!!.possibleAnswers
+        val colorPosition = Vector2(TextureSizes.distanceBetweenButtons, TextureSizes.distanceBetweenButtons)
+        for (i in colorOptions.indices) {
+            colorOptions[i] = ColorButton("", colorPosition, TextureSizes.colorButton, stringOfColors[i])
+            ui.addActor(colorOptions[i])
+            if (i % 2 == 0) {
+                colorPosition.x += TextureSizes.colorButton.width + TextureSizes.distanceBetweenButtons
+            } else {
+                colorPosition.y += TextureSizes.colorButton.height + TextureSizes.distanceBetweenButtons
+                colorPosition.x = TextureSizes.distanceBetweenButtons
+            }
+        }
+        var colorClicked = false
+        for (i in colorOptions.indices) {
```

Comment:
I propose that we use a more functional approach for this, and try to separate some logic into smaller functions.

```kotlin
colorOptions.indices.forEach { button ->
  button.onClick {
    // Place this method in TaskScreen
   // Let it find all other buttons and change their colors
    focusOnButton(button)
  }
}

// ... further down in TaskScreen
fun focusOnButton(button: Button) {
  // I use filterNotNull since items in colorOptions can be null.
  // It will guarantee that all items in otherColors are non-null
  val otherColors = colorOptions.filterNotNull { it != button }

  otherColors.forEach {
    it.changeColor("#737373")
  }
}
```